### PR TITLE
Improve readability of SNAP calcs

### PR DESCRIPTION
**Pull request by knod**
And adds documentation

### CODE REVIEW

File: src/programs/federal/snap.js [hunk 1]
```diff
@@ -49,140 +51,170 @@ const getSNAPBenefits = function (client, timeframe) {
...
+ *     to mess around with actual client values.
+ * 
+ * @namespace
+ *
+ * @note Almost all require the `current` or `future`
+ *     prop of the typical `client` object as an
+ *     argument.
+ */
 var SNAPhelpers = {},
     hlp         = SNAPhelpers;
```
Comment (ethanbb) [inline src/programs/federal/snap.js]: While we're making this file clearer: could we move these declarations up above `getSNAPBenefits`? I was scratching my head trying to figure out where `hlp` was coming from.

File: src/programs/federal/snap.js [hunk 2]
```diff
@@ -12,30 +12,32 @@ import { getLimitBySize } from '../../utils/getGovData';
...
+      netIncomeTestResult     = hlp.passesNetIncomeTest(client),
+      maxSnapAllotment        = getLimitBySize(SNAPData.SNAP_LIMITS, householdSize),
+      percentageOfNetIncome   = hlp.getNetIncome(client) * SNAPData.PERCENT_OF_NET,
       maxClientAllotment      = Math.max(0, maxSnapAllotment - percentageOfNetIncome);
 
   if (grossIncomeTestResult === true && netIncomeTestResult === true) {
 
     if (maxClientAllotment <= SNAPData.SMALL_HOUSEHOLD_MIN_GRANT) {
-      if (hlp.householdSize(client) <= SNAPData.SMALL_HOUSEHOLD_SIZE) {
+      if (householdSize <= SNAPData.SMALL_HOUSEHOLD_SIZE) {
```
Comment (ethanbb) [inline src/programs/federal/snap.js:54]: Currently, with this nested if statement, a household larger than `SMALL_HOUSEHOLD_SIZE` but with a max client allotment of less than `SMALL_HOUSEHOLD_MIN_GRANT` will get a `finalAmount` of 0. If that's a mistake, we should change this to a single if statement with an &&.

File: src/programs/federal/snap.js [hunk 3]
```diff
@@ -12,30 +12,32 @@ import { getLimitBySize } from '../../utils/getGovData';
...
-  getUnder13OfHousehold,
+  getYoungerThan,
+  getOlderThan,
 } from '../../utils/getMembers';
 
 
-/** Based on https://www.masslegalservices.org/SNAPCalculator */
-const getSNAPBenefits = function (client, timeframe) {
+/** Based on https://www.masslegalservices.org/SNAPCalculator
+ * @note Bay State CAP not included as this prototype only deals with
```
Comment (ethanbb) [inline src/programs/federal/snap.js]: I don't think `@note` is a real thing in `jsdoc` - do you see it in the output?

**Review by ethanbb — COMMENTED** [review]
OK this is pretty long. I just looked at the top of the first file so far. Will come back.

File: src/programs/federal/snap.js (hunk 1 above)
Comment (knod) [reply src/programs/federal/snap.js]: Good idea!

File: src/programs/federal/snap.js (hunk 3 above)
Comment (knod) [reply src/programs/federal/snap.js]: Dagnabit, it doesn't. I'll have to remove that wherever it is in the comments in general. It should be! And @warning should be too!

File: src/programs/federal/snap.js (hunk 2 above)
Comment (knod) [reply src/programs/federal/snap.js:54]: Not sure, but that's more of a problem with the logic and not this PR, right?

Comment (ethanbb) [reply src/programs/federal/snap.js:54]: Yeah true, I've just never read through this file before and thought it seemed important.

Comment (knod) [reply src/programs/federal/snap.js:54]: I don't remember any particular one of these, but I do remember finding several places where the logic didn't make sense to me and ended up matching the mass legal logic. We can take another look at that later for sure. @todo.

Other than that, are more changes needed?

**Comment by knod** [comment]
Because of issues with Travis and doing PRs from the same source branch onto multiple upstream branches, have to close and re-open a new PR for this. Will mention this one in that one to keep them linked, but the comments will not carry over.

**Pull request closed without merging**